examples/templates/azure-linux/main.tf

```diff
@@ -0,0 +1,227 @@
+terraform {
+  required_providers {
+    coder = {
+      source  = "coder/coder"
+      version = "0.4.3"
+    }
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = "=3.0.0"
+    }
+  }
+}
+
+variable "location" {
+  description = "What location should your workspace live in?"
+  default     = "eastus"
+  validation {
+    condition = contains([
+      "eastus",
+      "southcentralus",
+      "westus2",
+      "australiaeast",
+      "southeastasia",
+      "northeurope",
+      "westeurope",
+      "centralindia",
+      "eastasia",
+      "japaneast",
+      "brazilsouth",
+      "asia",
+      "asiapacific",
+      "australia",
+      "brazil",
+      "india",
+      "japan",
+      "southafrica",
+      "switzerland",
+      "uae",
+    ], var.location)
+    error_message = "Invalid location!"
+  }
+}
+
+variable "instance_type" {
+  description = "What instance type should your workspace use?"
+  default     = "Standard_B4ms"
+  validation {
+    condition = contains([
+      "Standard_B1ms",
+      "Standard_B2ms",
+      "Standard_B4ms",
+      "Standard_B8ms",
+      "Standard_B12ms",
+      "Standard_B16ms",
+      "Standard_D2as_v5",
+      "Standard_D4as_v5",
+      "Standard_D8as_v5",
+      "Standard_D16as_v5",
+      "Standard_D32as_v5",
+    ], var.instance_type)
+    error_message = "Invalid instance type!"
+  }
+}
+
+variable "home_size" {
+  type        = number
+  description = "How large would you like your home volume to be (in GB)?"
+  default     = 20
+  validation {
+    condition     = var.home_size >= 1
+    error_message = "Value must be greater than or equal to 1."
+  }
+}
+
+provider "azurerm" {
+  features {}
+}
+
+data "coder_workspace" "me" {
+}
+
+resource "coder_agent" "main" {
+  arch = "amd64"
+  os   = "linux"
+}
+
+locals {
+  prefix = "coder-${data.coder_workspace.me.owner}-${data.coder_workspace.me.name}"
+
+  userdata = templatefile("cloud-config.yaml.tftpl", {
+    username          = lower(substr(data.coder_workspace.me.owner, 0, 32))
+    init_script       = base64encode(coder_agent.main.init_script)
+    coder_agent_token = coder_agent.main.token
```

Comment:
It's frowned upon to expose secrets via userdata in Google Cloud and AWS, since that's generally publicly viewable through the dashboard. I'm not sure if it's similar in Azure, but I'd prefer if we used the Azure Instance Identity for authentication.

Reply:
I would prefer that too, but #3315

Reply:
Can you add a TODO with a reference to the issue?

Reply:
@spikecurtis that is a one-line fix. I'm happy to do it, but it's presently assigned to you.

Reply:
I’m happy for you to reassign! Or, whomever gets a moment first

Reply:
https://github.com/coder/coder/pull/3355 😎

Reply:
@spikecurtis I'm not sure what to do about this. The resource being used doesn't expose the instance ID, so I guess we just have to use token auth here?

Reply:
Yeah, unfortunately. Alternative is to unpick the home volume stuff, but that feels like a bigger loss.

Reply:
I'm trying a workaround described https://github.com/hashicorp/terraform-provider-azurerm/issues/6117